varats-core/varats/mapping/commit_map.py

```diff
@@ -29,11 +28,73 @@ class AmbiguousCommitHash(Exception):
 class CommitMap():
     """Provides a mapping from commit hash to additional information."""
 
-    def __init__(self, stream: tp.Iterable[str]) -> None:
-        self.__hash_to_id: CharTrie = CharTrie()
-        for line in stream:
-            slices = line.strip().split(', ')
-            self.__hash_to_id[slices[1]] = int(slices[0])
+    def __init__(
+        self,
+        git_path: Path,
+        end: str = "HEAD",
+        start: tp.Optional[str] = None,
+        refspec: str = "HEAD"
+    ) -> None:
+        self.git_path = git_path
+        self.end = end
+        self.start = start
+        self.refspec = refspec
+        self._hash_to_id = None
+        self._hash_to_id_master = None
+
+    @property
+    def __hash_to_id(self) -> CharTrie:
+        if not self._hash_to_id:
+            self._hash_to_id = self.generate_hash_to_id()
+        return self._hash_to_id
+
+    @property
+    def __hash_to_id_master(self) -> CharTrie:
+        if not self._hash_to_id_master:
+            self._hash_to_id_master = self.generate_hash_to_id(master=True)
+        return self._hash_to_id_master
+
+    def generate_hash_to_id(self, master: bool = False) -> CharTrie:
+        search_range = ""
+        if self.start is not None:
+            search_range += self.start + ".."
+        search_range += self.end
+
+        with local.cwd(self.git_path):
+            old_head = get_current_branch()
+            git("checkout", self.refspec)
+            if master:
+                full_out = git(
+                    "--no-pager", "log", "--all", "--pretty=format:'%H'"
+                )
+                wanted_out = git(
+                    "--no-pager", "log", "--all", "--pretty=format:'%H'",
+                    search_range
+                )
+            else:
+                full_out = git("--no-pager", "log", "--pretty=format:'%H'")
+                wanted_out = git(
+                    "--no-pager", "log", "--pretty=format:'%H'", search_range
+                )
```

Comment:
Shouldn't the commands look like this?

```suggestion
            if master:
                full_out = git(
                    "--no-pager", "log", "--all", "--pretty=format:'%H'"
                )
                wanted_out = git(
                    "--no-pager", "log", "--pretty=format:'%H'",
                    search_range
                )
            else:
                full_out = git("--no-pager", "log", "--all", "--pretty=format:'%H'")
                wanted_out = git(
                    "--no-pager", "log", "--all", "--pretty=format:'%H'", search_range
                )
```

Reply:
Yes, you're absolutely right, I somehow mixed that up, thanks! I fixed it in a new commit.